optimum/exporters/openvino/model_patcher.py

```diff
@@ -6754,3 +6754,87 @@ def __exit__(self, exc_type, exc_value, traceback):
         setattr(self._model, self.orig_forward_name, self.orig_forward)
         for layer in self._model.backbone.layers:
             layer.mixer.forward = layer.mixer._orig_forward
+
+
+# This patcher class serves for exporting Zamba2 model to OpenVINO IR
+class Zamba2ModelPatcher(ModelPatcher):
+    def __init__(
+        self,
+        config: "OnnxConfig",
+        model: Union["PreTrainedModel", "TFPreTrainedModel"],
+        model_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        super().__init__(config, model, model_kwargs)
+
+        from transformers.models.zamba2.modeling_zamba2 import Zamba2HybridDynamicCache
+
+        class Zamba2HybridDynamicCacheWrap(Zamba2HybridDynamicCache):
+            def __init__(self, config, batch_size: int, conv_states, ssm_states, key_cache, value_cache):
+                # Call parent constructor with all required arguments
+                super().__init__(config=config, batch_size=batch_size)
+                self.conv_states = conv_states
+                self.ssm_states = ssm_states
+                self.key_cache = key_cache
+                self.value_cache = value_cache
+
+        # the patch is needed to include KV-cache, Conv, and SSM states in the inputs and outputs.
+        def patched_forward(
+            input_ids,
+            attention_mask=None,
+            position_ids=None,
+            # cache_position=None,
+            past_key_values=None,
+        ):
+            num_hidden_layers = self.real_config._config.num_hidden_layers
+            use_cache = False
+            wrapped_cache_params = None
+            if past_key_values is not None:
+                use_cache = True
+                conv_states = []
+                ssm_states = []
+                key_cache = []
+                value_cache = []
+                # inputs passed in an order of (key, value, conv_state, ssm_state)
```

Comment:
ok so length of past_key_values is always 4 * num_hidden_layers is this correct ? if yes would you mind adding ?